xonsh/completer.py

```diff
@@ -17,6 +17,35 @@
 RE_ATTR = re.compile(r'(\S+(\..+)*)\.(\w*)$')
 RE_WIN_DRIVE = re.compile(r'^([a-zA-Z]):\\')
 
+
+def _path_from_partial_string(inp, pos=None):
+    if pos is None:
+        pos = len(inp)
+    partial = inp[:pos]
+    x, b = partial_string_finder(partial)
+    lx = len(x)
+    if lx == 0:
+        return None
+    elif lx % 2: # odd number: in the middle of a string
+        s = partial[x[-1]:]
+    else:
+        if x[-1] != pos:
+            return None
+        s = partial[x[-2]:x[-1]]
+    b = b[-1]
+    end = re.sub('[rRbBuU]*','',b)
+    _s = s
+    if not _s.endswith(end):
+        _s = _s + end
+    try:
+        val = eval(_s)
+    except:
+        return None
+    if isinstance(val, bytes):
+        val = val.decode()
```

Comment:
I think all decode() calls need to get passed `$XONSH_ENCODING` and `$XONSH_ENCONDING_ERRORS` in order to safely handle various unicode issues.  Given the purpose of this function, you might want to pass these values in instead of looking them up here.